File: lib/models/task.dart
class Task {
  int? id;
  String title;
  String description;
  int? categoryId;
  String priority;
  String? dueDate;
  bool isCompleted;
  String? createdAt;
  String? updatedAt;
  String? categoryName;
  String? categoryColor;

  Task({
    this.id,
    required this.title,
    required this.description,
    this.categoryId,
    this.priority = 'medium',
    this.dueDate,
    this.isCompleted = false,
    this.createdAt,
    this.updatedAt,
    this.categoryName,
    this.categoryColor,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'] ?? '',
      categoryId: json['category_id'],
      priority: json['priority'] ?? 'medium',
      dueDate: json['due_date'],
      isCompleted: json['is_completed'] == 1,
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
      categoryName: json['category_name'],
      categoryColor: json['category_color'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'category_id': categoryId,
      'priority': priority,
      'due_date': dueDate,
      'is_completed': isCompleted,
    };
  }
}

class Category {
  int? id;
  String name;
  String color;

  Category({
    this.id,
    required this.name,
    required this.color,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      color: json['color'],
    );
  }
}

File: task_manager _api/models/Task.php
<?php
class Task {
    private $conn;
    private $table_name = "tasks";

    public $id;
    public $title;
    public $description;
    public $category_id;
    public $priority;
    public $due_date;
    public $is_completed;
    public $created_at;
    public $updated_at;

    public function __construct($db) {
        $this->conn = $db;
    }

    public function read() {
        $query = "SELECT t.*, c.name as category_name, c.color as category_color 
                  FROM " . $this->table_name . " t 
                  LEFT JOIN categories c ON t.category_id = c.id 
                  ORDER BY t.created_at DESC";
        
        $stmt = $this->conn->prepare($query);
        $stmt->execute();
        
        return $stmt;
    }

    public function create() {
        $query = "INSERT INTO " . $this->table_name . " 
                  SET title=:title, description=:description, category_id=:category_id, 
                  priority=:priority, due_date=:due_date";
        
        $stmt = $this->conn->prepare($query);
        
        $stmt->bindParam(":title", $this->title);
        $stmt->bindParam(":description", $this->description);
        $stmt->bindParam(":category_id", $this->category_id);
        $stmt->bindParam(":priority", $this->priority);
        $stmt->bindParam(":due_date", $this->due_date);
        
        if($stmt->execute()) {
            return true;
        }
        return false;
    }

    public function update() {
        $query = "UPDATE " . $this->table_name . " 
                  SET title=:title, description=:description, category_id=:category_id, 
                  priority=:priority, due_date=:due_date, is_completed=:is_completed 
                  WHERE id=:id";
        
        $stmt = $this->conn->prepare($query);
        
        $stmt->bindParam(":title", $this->title);
        $stmt->bindParam(":description", $this->description);
        $stmt->bindParam(":category_id", $this->category_id);
        $stmt->bindParam(":priority", $this->priority);
        $stmt->bindParam(":due_date", $this->due_date);
        $stmt->bindParam(":is_completed", $this->is_completed);
        $stmt->bindParam(":id", $this->id);
        
        if($stmt->execute()) {
            return true;
        }
        return false;
    }

    public function delete() {
        $query = "DELETE FROM " . $this->table_name . " WHERE id = ?";
        $stmt = $this->conn->prepare($query);
        $stmt->bindParam(1, $this->id);
        
        if($stmt->execute()) {
            return true;
        }
        return false;
    }
}
?>